modules/java/generator/src/java/android+JavaCameraView.java

```diff
@@ -303,7 +311,10 @@ public Mat gray() {
 
         @Override
         public Mat rgba() {
-            Imgproc.cvtColor(mYuvFrameData, mRgba, Imgproc.COLOR_YUV2RGBA_NV21, 4);
+            if (mPreviewFormat == ImageFormat.NV21)
+                Imgproc.cvtColor(mYuvFrameData, mRgba, Imgproc.COLOR_YUV2RGBA_NV21, 4);
+            else if (mPreviewFormat == ImageFormat.YV12)
+                Imgproc.cvtColor(mYuvFrameData, mRgba, Imgproc.COLOR_YUV2RGB_I420, 4);  // COLOR_YUV2RGBA_YV12 produces inverted colors
```

Comment:
Is there a need for an else branch? On initialisation, the previewFormat is always set to NV21 or YV12:

```
if (Build.BRAND.equalsIgnoreCase("generic") || Build.BRAND.equalsIgnoreCase("Android"))
    params.setPreviewFormat(ImageFormat.YV12);  // "generic" or "android" = android emulator
else
    params.setPreviewFormat(ImageFormat.NV21);
```

I still kept the " if (mPreviewFormat == ImageFormat.YV12)" condition because I thought it looked more informative, but that part can be safely removed.